### PR TITLE
Move ISR code into a seperate file and make sure it exist in main ROM

### PR DESCRIPTION
**Pull request by howard0su**

### CODE REVIEW

File: src/target/common/devo/pwm_isr.c [hunk 1]
```diff
@@ -12,20 +12,20 @@
...
-
 #include <libopencm3/stm32/rcc.h>
 #include <libopencm3/stm32/timer.h>
 #include <libopencm3/stm32/gpio.h>
 #include <libopencm3/stm32/dma.h>
 #include <libopencm3/cm3/nvic.h>
 
+#include "common.h"
+#ifndef DISABLE_PWM
+
```
Comment (PhracturedBlue) [inline src/target/common/devo/pwm_isr.c]: Is there a reason for these changes other than stylistic?

File: src/target/common/devo/protocol/pwm.c [hunk 2]
```diff
@@ -79,6 +79,7 @@ void PWM_Stop()
 #endif
 }
 
+extern volatile u16 *pwm; // defined in pwm_isr.c
```
Comment (hexfet) [inline src/target/common/devo/protocol/pwm.c]: Variable pwm is unused.  It was removed when #472 was merged.

File: src/target/common/devo/pwm_isr.c [hunk 3]
```diff
@@ -12,20 +12,20 @@
...
 #include "../ports.h"
 
+__attribute__((__used__)) volatile u16 *pwm;
```
Comment (hexfet) [inline src/target/common/devo/pwm_isr.c]: Variable pwm is unused.

**Comment by hexfet** [comment]
Please also change the ISRs in target/common/devo/soft_serial.c, or if you'd like I can make a try.

**Comment by howard0su** [comment]
@hexfet good catch. I am not sure why I missed this.

**Pull request merged**